src/segment-loader.js

```diff
@@ -717,6 +719,91 @@ export default class SegmentLoader extends videojs.EventTarget {
     };
   }
 
+  abortRequestEarly_(stats) {
+    // Do not consider aborting early if the player is paused
+    if (this.hls_.tech_.paused() ||
+        // Or if the current playlist is on the lowestEnabledRendition
+        // TODO: Replace timeout with a boolean indicating whether this playlist is the
+        //       lowestEnabledRendition.
+        !this.xhrOptions_.timeout ||
+        // Or if the the current playlist does not have bandwidth information that can
+        // be used in the estimation calculations below
+        !(this.playlist_.attributes && this.playlist_.attributes.BANDWIDTH)) {
+      return false;
+    }
+
+    // Wait at least 1 second since the first byte of data has been received before
+    // using the calculated bandwidth from the] progress event to allow the bitrate
+    // to stabilize
+    if (Date.now() - stats.firstByteRoundTripTime < 1000) {
+      return false;
+    }
+
+    const measuredBandwidth = stats.bandwidth;
+    const estimatedSegmentSize =
+      this.pendingSegment_.duration * this.playlist_.attributes.BANDWIDTH;
+    const requestTimeRemaining =
+      (estimatedSegmentSize - (stats.bytesReceived * 8)) / measuredBandwidth;
+    const buffered = this.buffered_();
+    const bufferedEnd = buffered.length ? buffered.end(buffered.length - 1) : 0;
+    const timeUntilRebuffer =
+      (bufferedEnd - this.currentTime_()) / this.hls_.tech_.playbackRate();
+
+    // Only consider aborting early if the estimated time to finish the download
+    // is larger than the estimated time until the player runs out of forward buffer
+    if (requestTimeRemaining < timeUntilRebuffer) {
+      return false;
+    }
+
+    const { playlist, roundTripTime } =
+      minRebufferingSelector(this.hls_.playlists.master,
+                             measuredBandwidth,
+                             timeUntilRebuffer,
+                             this.pendingSegment_.duration);
+
+    let minimumTimeSaving = 0.5
+
+    // If we are already rebuffering, increase the amount of variance we add to the
+    // potential round trip time of the new request so that we are not too agressive
+    // with switching to a playlst that might save us a fraction of a second.
+    if (timeUntilRebuffer <= TIME_FUDGE_FACTOR) {
+      minimumTimeSaving = 1;
+    }
+
+    // Finally, don't abort early and trigger the downswitch if the playlist
```

Comment:
Can remove the comments around the conditionals here, since they don't add much over reading the code itself.